webknossos/webknossos/dataset/dataset.py

```diff
@@ -441,6 +505,178 @@ def download(
+        ]
+        if len(input_files) == 0:
+            raise ValueError(
+                "Could not find any image data supported. "
+                + f"The following suffixes are supported: {sorted(valid_suffixes)}"
+            )
+
+        if (
+            map_filepath_to_layer_name
+            == Dataset.ConversionLayerMapping.ENFORCE_LAYER_PER_FILE
+        ):
+            map_filepath_to_layer_name = str
+        elif (
+            map_filepath_to_layer_name
+            == Dataset.ConversionLayerMapping.ENFORCE_SINGLE_LAYER
+        ):
+            map_filepath_to_layer_name = lambda p: input_upath.name
+        elif (
+            map_filepath_to_layer_name
+            == Dataset.ConversionLayerMapping.ENFORCE_LAYER_PER_FOLDER
+        ):
+            map_filepath_to_layer_name = (
+                lambda p: input_upath.name if p.parent == Path() else str(p.parent)
+            )
+        elif (
+            map_filepath_to_layer_name
+            == Dataset.ConversionLayerMapping.ENFORCE_LAYER_PER_TOPLEVEL_FOLDER
+        ):
+            map_filepath_to_layer_name = (
+                lambda p: input_upath.name if p.parent == Path() else p.parts[0]
+            )
+        elif (
+            map_filepath_to_layer_name
+            == Dataset.ConversionLayerMapping.INSPECT_EVERY_FILE
+        ):
+            # If a file has z dimensions, it becomes it's own layer,
+            # if it's 2D, the folder becomes a layer.
+            map_filepath_to_layer_name = (
+                lambda p: str(p)
+                if _has_image_z_dimension(
+                    input_upath / p,
+                    use_bioformats=use_bioformats,
+                    is_segmentation=_guess_if_segmentation_path(p),
+                )
+                else input_upath.name
+                if p.parent == Path()
+                else str(p.parent)
+            )
+        elif (
+            map_filepath_to_layer_name
+            == Dataset.ConversionLayerMapping.INSPECT_SINGLE_FILE
+        ):
+            # As before, but only a single image is inspected to determine 2D vs 3D.
+            if _has_image_z_dimension(
+                input_path / input_files[0],
+                use_bioformats=use_bioformats,
+                is_segmentation=_guess_if_segmentation_path(input_files[0]),
+            ):
+                map_filepath_to_layer_name = str
+            else:
+                map_filepath_to_layer_name = (
+                    lambda p: input_upath.name if p.parent == Path() else str(p.parent)
+                )
+        elif isinstance(map_filepath_to_layer_name, Dataset.ConversionLayerMapping):
+            raise ValueError(
+                f"Got unexpected ConversionLayerMapping value for map_filepath_to_layer_name: {map_filepath_to_layer_name}"
+            )
+
+        ds = cls(output_path, voxel_size=voxel_size, name=name)
+
+        filepaths_per_layer: Dict[str, List[Path]] = {}
```

Comment:
```suggestion
        file_paths_per_layer: Dict[str, List[Path]] = {}
```

Reply:
Then I should also rename `map_filepath_to_layer_name` to `map_file_path_to_layer_name`, right? Also possible, not sure which one I like more…

Reply:
Alright I’ll take that back, apparently there is an ongoing process of including the word filepath into the english language, let’s leave it as you wrote it, then